runtime/kubeadm.go

```diff
@@ -59,21 +60,41 @@ func (d *Default) templateFromContent(templateContent string) ([]byte, error) {
 	envMap[APIServer] = d.APIServer
 	envMap[PodCIDR] = d.PodCIDR
 	envMap[SvcCIDR] = d.SvcCIDR
+	envMap[KubeadmAPI] = d.KubeadmAPI
+	envMap[CriSocket] = d.CriSocket
+	// we need to Dynamic get cgroup driver on init master.
+	envMap[CriCGroupDriver] = d.CriCGroupDriver
 	envMap[Repo] = fmt.Sprintf("%s:%d/library", SeaHub, d.RegistryPort)
 	envMap[EtcdServers] = getEtcdEndpointsWithHTTPSPrefix(d.Masters)
 	var buffer bytes.Buffer
 	err = tmpl.Execute(&buffer, envMap)
 	return buffer.Bytes(), err
 }
 
-func (d *Default) kubeadmConfig() string {
-	var sb strings.Builder
+// setKubeadmAPIByVersion is set kubeadm api version and crisocket
+func (d *Default) setKubeadmAPIByVersion() {
+	switch {
```

Comment:
if people want to custom crisocket. this is a not good method . 
eg. kubernetes 1.21.2 , I just want to use docker . but sealer use containerd. this is annoying..